editor/inspector/editor_preview_plugins.h

```diff
@@ -67,10 +68,55 @@ class EditorBitmapPreviewPlugin : public EditorResourcePreviewGenerator {
 class EditorPackedScenePreviewPlugin : public EditorResourcePreviewGenerator {
 	GDCLASS(EditorPackedScenePreviewPlugin, EditorResourcePreviewGenerator);
 
+	mutable Mutex *abort_mutex;
+	mutable Mutex *construct_mutex;
+	mutable bool aborted = false;
+	mutable DrawRequester draw_requester;
+	mutable HashMap<String, HashMap<String, int>> node_lookup_tables;
+	mutable bool scene_construct_done;
+	mutable RID vp_3d_rid;
+	mutable RID vp_2d_rid;
+	mutable RID vp_gui_rid;
+	mutable SubViewport *p_vp_3d;
+	mutable SubViewport *p_vp_2d;
+	mutable SubViewport *p_vp_gui;
+
+protected:
+	// --- 3D preview ---
+	void _setup_scene_3d(Node *p_node) const;
+	void _construct_scene_3d(Ref<PackedScene> p_pack, const Size2 &p_size, int p_light_count) const;
+
+	void _calculate_scene_aabb(Node *p_node, AABB &p_aabb) const;
+	Transform3D _get_global_transform_3d(Node *p_n3d) const;
+
+	// --- 2D preview ---
+	void _setup_scene_2d(Node *p_node) const;
+	void _construct_scene_2d(Ref<PackedScene> p_pack, const Size2 &p_size) const;
+
+	void _calculate_scene_rect(Node *p_node, Rect2 &r_rect) const;
+	void _hide_node_2d_in_scene(Node *p_node) const;
+	void _hide_gui_in_scene(Node *p_node) const;
+
+	// Utils
+	bool _get_aborted() const;
+	void _set_scene_construct_done(bool done) const;
```

Comment:
```suggestion
	void _set_scene_construct_done(bool p_done) const;
```
Please prefix arguments with `p_`